src/main/java/controller/MainController.java

```diff
@@ -1,241 +1,331 @@
+		results = getCodeQualityResults();
+		String[] colNames = new String[5 + rulesList.size()];
+		colNames[0] = "Method ID";
+		colNames[1] = "long_method";
+		colNames[2] = "feature_envy";
+		colNames[3] = "PMD";
+		colNames[4] = "iPlasma";
+		int iterator = 5;
+		for (CodeQualityRule rule : rulesList) {
+			colNames[iterator] = rule.getName();
+			iterator++;
+		}
+
+		if (results != null) {
+			qualityGui.fillTable(results, colNames);
+			qualityGui.show();
+		}
+	}
+
+	/**
+	 * 
+	 * Returns the results of the calculation of each rule, for each method.
+	 * 
+	 * @return An Array of String arrays where each line is a row with the code
+	 *         quality results for a method, and each column is the value of that
+	 *         result line for that column
+	 */
+	private String[][] getCodeQualityResults() {
+		String[][] results = new String[excelRowsConverted.size()][5 + rulesList.size()];
+		int iterator = 0;
+		for (ExcelRow row : excelRowsConverted) {
+			String[] qualityRow = new String[4 + excelRowsConverted.size()];
+			qualityRow[0] = Integer.toString(row.getId());
+			qualityRow[1] = Boolean.toString(row.isIs_long_method());
+			qualityRow[2] = Boolean.toString(row.isIs_feature_envy());
+			qualityRow[3] = Boolean.toString(row.isPMD());
+			qualityRow[4] = Boolean.toString(row.isiPlasma());
+			int ruleIterator = 5;
+
+			CodeQualityRule wrongRule = null;
+			for (CodeQualityRule rule : rulesList) {
+				try {
+					qualityRow[ruleIterator] = getResult(rule, row);
+					ruleIterator++;
+				} catch (ScriptException e) {
+					qualityGui.hide();
+					wrongRule = rule;
+					JOptionPane.showMessageDialog(null,
+							"Invalid rule syntax! Please verify the rule with the name \"" + wrongRule + "\"!");
+					return null;
+				}
+			}
+			results[iterator] = qualityRow;
+			iterator++;
+		}
+		return results;
+
+	}
+
+	/**
+	 * Runs a rule over an excelRow and returns the result.
+	 * 
+	 * @param rule The rule, the result of which we require.
+	 * @param row  The excel row containing the methodID over which we wish to run
+	 *             the rule.
+	 * @return Returns the result of running the rule over the methodID of the given
+	 *         ExcelRow, in string form.
+	 * @throws ScriptException
+	 */
+	private String getResult(CodeQualityRule rule, ExcelRow row) throws ScriptException {
+		ScriptEngineManager engineManager = new ScriptEngineManager();
+		ScriptEngine engine = engineManager.getEngineByName("ECMAScript");
+		String filledRule = registerVariables(row);
+		Object result = null;
+		result = engine.eval(filledRule + "eval('" + rule.getRule() + "');");
+		return Boolean.toString(Boolean.TRUE.equals(result));
+	}
+
+	/**
+	 * Creates and returns a string ready to be passed on to a javascript engine,
+	 * which initializes all the necessary metric variables.
+	 * 
+	 * @param row The excel row with the values for our metrics.
+	 * @return filledRule The Sting of rule initializations to be passed on to a

```

Comment:
the Sting! oh noes! also, it doesn't return rule initializations. it returns the metrics turned into variables to use in the JS engine for running the rules